### DISCUSSION TITLE
Auto Layout with Sub Flow

### DISCUSSION THREAD

**Discussion by vaezc** [xyflow/xyflow#4185] [Layouting]
I am currently using a feature that utilizes the grouping functionality. I want to implement automatic layout, and I will be dynamically changing the number of nodes, similar to the autolayout pro example. However, I have added grouping functionality, and I've noticed that the automatic layout doesn't seem to take effect when groups are present. Would it be possible for the official team to provide an example demonstrating how to use autolayout with grouping?

**Comment by chrtze** [comment]
Hey @vaezc, thanks for your message. This is on our list and we already tried to create an example for this. However, most of the layout libraries don't support or don't work properly with sub flows (for example dagre: https://github.com/dagrejs/dagre/issues/238). The easiest way for now would be to adapt the elkjs code to support sub flows. Did you try that?

**Reply by vaezc** [reply]
Yes, I have already tried that. I am currently using the ELK subflow feature. My goal is to dynamically add nodes while utilizing subflows, but I've encountered an issue where the positions of existing nodes shift after new nodes are added. This behavior is not what I expected. The autolayout in the pro example is able to remember node positions, but I am unsure how to integrate it with subflows.

**Reply by vaezc** [reply]
excuse me，do you have any more news?
